relayer-engine/src/storage/storage.ts

```diff
@@ -165,46 +237,216 @@ export class DefaultStorage implements Storage {
       if (!key) {
         return null;
       }
-      const raw = nnull(await redis.get(key));
-      const workflow = JSON.parse(raw);
-      if (workflow.scheduledAt) {
-        workflow.scheduledAt = new Date(workflow.scheduledAt);
-      }
+      const raw = nnull(await redis.hGetAll(key));
+      const workflow = this.rawObjToWorkflow(raw);
+      const now = new Date();
+      await redis.hSet(key, {
+        processingBy: this.nodeId,
+        startedProcessingAt: now.getTime(),
+      });
       return { workflow, plugin: nnull(this.plugins.get(workflow.pluginName)) };
     });
   }
 
+  private rawObjToWorkflow(raw: Record<string, string>): Workflow {
+    return {
+      id: raw.id,
+      data: JSON.parse(raw.data),
+      completedAt: raw.completedAt ? new Date(raw.completedAt) : undefined,
+      scheduledAt: raw.scheduledAt ? new Date(raw.scheduledAt) : undefined,
+      failedAt: raw.failedAt ? new Date(raw.failedAt) : undefined,
+      startedProcessingAt: raw.startedProcessingAt
+        ? new Date(raw.startedProcessingAt)
+        : undefined,
+      scheduledBy: raw.scheduledBy,
+      processingBy: raw.processingBy,
+      pluginName: raw.pluginName,
+      retryCount: Number(raw.retryCount),
+      maxRetries: Number(raw.maxRetries),
+    };
+  }
+
   // Demote workflows from active set based off plugin config
   async handleStorageStartupConfig(plugins: Plugin[]): Promise<void> {
     this.logger.debug("Handling storage startup config");
     const pluginToShouldDemote = new Map(
       plugins.map(p => [p.pluginName, p.demoteInProgress]),
     );
     this.logger.info("Checking for inProgress workflows to demote on startup");
+    // try {
```

Comment:
What's your plan here? We can just remove this I think?

Reply:
I had a pending conversation with you about it, because if there are multiple executors I don't know how that would work, so I didn't want to remove it but I had to think about what we do with it or how you envisioned it working.

Reply:
We decided this is no longer necessary.